### PR TITLE
Generate redis

### PR DESCRIPTION
**Pull request by crwilcox**

### CODE REVIEW

**Comment by crwilcox** [comment]
Locally I have 8 failures. They all resemble this:

```
8) Error:
Google::Cloud::Redis::V1beta1::CloudRedisClient::delete_instance#test_0002_invokes delete_instance and returns an operation error.:
Google::Gax::RetryError: GaxError Exception occurred in retry method that was not classified as transient, caused by undefined method `execute' for #<Google::Longrunning::Operation:0x00007fe3a8028058>
    /usr/local/Cellar/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/google-gax-1.2.0/lib/google/gax/api_callable.rb:369:in `rescue in block in retryable'
    /usr/local/Cellar/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/google-gax-1.2.0/lib/google/gax/api_callable.rb:358:in `block in retryable'
    /usr/local/Cellar/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/google-gax-1.2.0/lib/google/gax/api_callable.rb:230:in `block in create_api_call'
    /usr/local/Cellar/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/google-gax-1.2.0/lib/google/gax/api_callable.rb:260:in `block in create_api_call'
    /Users/crwilcox/workspace/google-cloud-ruby/google-cloud-redis/lib/google/cloud/redis/v1beta1/cloud_redis_client.rb:565:in `delete_instance'
    /Users/crwilcox/workspace/google-cloud-ruby/google-cloud-redis/test/google/cloud/redis/v1beta1/cloud_redis_client_test.rb:631:in `block (5 levels) in <top (required)>'
```

**Comment by crwilcox** [comment]
CI is passing locally. I had to make the following hand edits:
- Require path was 'doubled' https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/2078/commits/38a48ffdf421e0fa2d61749d85bdd5fc26510a52
- google-gax dependency version was wrong https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/2078/commits/e90685e1d879db704609c33c0f43f35203e1dc53

**Comment by crwilcox** [comment]
@jbolinger, can you explain why doing the manual change you suggest is preferable to rev'ing the version of google-gax? This did resolve issues under test and appears to be done in #2020 so I am trying to understand the different approaches.

**Comment by jbolinger** [comment]
@crwilcox the manual change isn't preferable to fix the tests - sorry if that part was misleading. We just need to cut a new release so that you don't have to do it manually to be compatible with the latest gax version.

The other issue with the duplicate path is an issue that I haven't seen before and am looking into.

**Comment by crwilcox** [comment]
@jbolinger are any changes needed to merge this? The test run is broken but you have a bug for that.

**Comment by blowmage** [comment]
The tests need to be updated to match the new google-gax implementation. I'll get them updated and get CI passing.

**Comment by crwilcox** [comment]
Consenting to other author commits to appease @googlebot

File: google-cloud-redis/google-cloud-redis.gemspec [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Gem::Specification.new do |gem|
+  gem.name          = "google-cloud-redis"
+  gem.version       = "0.1.0"
+
+  gem.authors       = ["Google LLC"]
+  gem.email         = "googleapis-packages@google.com"
+  gem.description   = "google-cloud-redis is the official library for Google Cloud Memorystore for Redis API."
+  gem.summary       = "API Client library for Google Cloud Memorystore for Redis API"
+  gem.homepage      = "https://github.com/googleapis/googleapis"
```
Comment (jbolinger) [inline google-cloud-redis/google-cloud-redis.gemspec]: We should also update the homepage (example: https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-dlp/google-cloud-dlp.gemspec)

File: google-cloud-redis/Gemfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+source "https://rubygems.org"
+
+gemspec
+
+gem "gcloud-jsondoc",
+    git: "https://github.com/GoogleCloudPlatform/google-cloud-ruby.git",
+    branch: "gcloud-jsondoc"
+
+# WORKAROUND: builds are having problems since the release of 3.0.0
```
Comment (jbolinger) [inline google-cloud-redis/Gemfile]: We should probably add this additional workaround to new libraries like this one (https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-dlp/Gemfile#L9). 

I don't actually know the history of this, perhaps @blowmage can explain

File: google-cloud-redis/google-cloud-redis.gemspec (hunk 1 above)
Comment (blowmage) [reply google-cloud-redis/google-cloud-redis.gemspec]: I'll make this change.

File: google-cloud-redis/Gemfile (hunk 2 above)
Comment (blowmage) [reply google-cloud-redis/Gemfile]: Verion 3.0.0 of [public_suffix](https://rubygems.org/gems/public_suffix) dropped support for Ruby 2.0. We occasionally see the build becoming confused by resolving dependencies for a different Ruby version than it is using.

**Pull request merged**